spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -141,5 +174,90 @@ module.exports = function (provider, setup) {
     });
   });
 
-  // TODO: add tests for tcpsocket.secure, accepting multiple.
+  it("Secures a socket", function (done) {
+    socket.connect('www.google.com', 443, function () {
+      var prepared;
+      // prepareSecure is Chrome specific, see freedom-for-chrome
+      socket.prepareSecure(function () {
+        prepared = true;
+      });
+      // TODO: timeout approach sucks, preferred approach:
```

Comment:
I think the issue is that preparesecure / secure isn't structured for HTTPS
connections - it was added as a hack for chrome to handle xmpp traffic.  if
you're securing immediately, you don't need the prepareSecure stuff, just
that if you have a protocol like xmpp that uses STARTSSL to upgrade
mid-connection, then you do.

so either, just use secure and don't worry about prepareSecure for now, or
the test will need to simulate a protocol that does ssl upgrading.

On Mon, Mar 23, 2015 at 3:15 PM, soycode <notifications@github.com> wrote:

> In spec/providers/coreIntegration/tcpsocket.integration.src.js
> <https://github.com/freedomjs/freedom/pull/241#discussion_r26986251>:
>
> > @@ -141,5 +174,90 @@ module.exports = function (provider, setup) {
> >      });
> >    });
> >
> > -  // TODO: add tests for tcpsocket.secure, accepting multiple.
> > +  it("Secures a socket", function (done) {
> > +    socket.connect('www.google.com', 443, function () {
> > +      var prepared;
> > +      // prepareSecure is Chrome specific, see freedom-for-chrome
> > +      socket.prepareSecure(function () {
> > +        prepared = true;
> > +      });
> > +      // TODO: timeout approach sucks, preferred approach:
>
> So this test is still not ideal - I can't get the dispatch object to
> receive the expected onData message per
> https://github.com/freedomjs/freedom/wiki/prepareSecure-API-Usage and so
> instead put in a timeout for now. It's also not clear whether or not the
> socket is even truly connecting, maybe there's a better test target out
> there than google.com:443.
>
> Anyway, all this *really* tests for now is that prepareSecure and secure
> run and execute callbacks, not that they really do what we want. Definitely
> want to improve this, suggestions are welcome, for now I'll switch to
> another problem.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freedomjs/freedom/pull/241/files#r26986251>.
>